core/src/main/java/io/questdb/cutlass/pgwire/modern/PGPipelineEntry.java

```diff
@@ -272,13 +273,13 @@ public void close() {
         if (!isCopy) {
             // if we are a copy, we do not own operations -> we cannot close them
             // so we just null them out and let the original entry close them
-            insertOp = Misc.free(insertOp);
+            tai = Misc.free(tai);
```

Comment:
Can you please explain the logic behind this change? 

I am only just starting to review it, but I read it like this:
When closing an entry (e.g. when returning the entry back to a pool) this change `free()` the entire INSERT cache. This looks add as the INSERT cache lifecycle is bound to a lifecycle of a session (context), not to a life-cycle of an individual entry. What am I missing?

Reply:
Isn't `tai` just one cache entry, and `taiCache` the whole cache?

Reply:
But, since `insertOp` is the only closeable thing in `tai`, this wouldn't avoid any memory leak.

Reply:
@mtopolnik tai is a single insert + bind var types; taiCache is the associative cache for tai objects; taiPool is an object pool for tai objects.

Reply:
@jerrinot the problem here was that we were closing insert operation without closing its tai object. When tai object is closed, it's returned to taiPool. So, we should close tai instead of closing the insert operation directly. Otherwise, if we don't close tai, it'll contain the closed insert operation leading to use-after-free. Hope this makes sense.

Reply:
> this wouldn't avoid any memory leak

@mtopolnik this change fixes use-after-free, not mem leak. The problem is twofold: we never closed insert op's SQL function, but also our tai cache/pool manipulations were problematic.